app/src/ui/lib/aria-types.ts

```diff
@@ -0,0 +1,16 @@
+export enum AriaHasPopup {
+  Dialog = 'dialog',
+  Menu = 'menu',
+  Listbox = 'listbox',
+  Tree = 'tree',
+  Grid = 'grid',
+}
+
+export type AriaHasPopupType =
+  | boolean
+  | AriaHasPopup.Dialog
+  | AriaHasPopup.Menu
+  | AriaHasPopup.Listbox
+  | AriaHasPopup.Tree
+  | AriaHasPopup.Grid
+  | undefined
```

Comment:
Another way to express this could be…
```suggestion
import { HTMLAttributes } from 'react'

export type AriaHasPopupType = HTMLAttributes<any>['aria-haspopup']
```